File: queue-stack/queue.go
package leetcode

type Queue struct {
	data  []int
	start int
}

func NewQueue() *Queue {
	data := make([]int, 0, 10)
	return &Queue{
		data:  data,
		start: 0,
	}
}

func (q *Queue) enQueue(x int) bool {
	q.data = append(q.data, x)
	return true
}

func (q *Queue) deQueue() bool {
	if q.isEmpty() {
		return false
	}

	q.start++
	return true
}

func (q *Queue) Front() int {
	return q.data[q.start]
}

func (q *Queue) isEmpty() bool {
	return q.start >= len(q.data)
}

File: queue-stack/Queue.java
class Queue {
	private List<Interger> data;
	private int start;
	public Queue() {
		this.data = new ArraryList<Interger>();
		this.start = 0;
	}

	public boolean isEmpty() {
		return this.start >= this.date.size();
	}

	public boolean enQueue(int x) {
		this.data.add(x);
		return true;
	}

	public boolea deQueue() {
		if (this.isEmpty()) {
			return false
		}

		this.start ++;
		return true;
	}

	public int Front() {
		return this.date.get(start);
	}
}
